### ISSUE TITLE
Update test files to matching naming convention

### ISSUE THREAD

**Issue by mschwager** [returntocorp/sgrep-rules#828]
Semgrep's `--test` functionality is detecting that the following rules don't have tests:

```
	python/django/best-practice/upsell_django_environ.yaml
	python/django/performance/access-foreign-keys.yaml
	go/lang/security/decompression_bomb.yaml
	go/lang/security/audit/crypto/use_of_weak_crypto.yaml
	go/lang/correctness/permissions/file_permission.yaml
	ruby/lang/security/yaml-parsing-with-version.yaml
```

However, e.g. we can see that they do have tests:

```
$ ls go/lang/security/audit/crypto/
bad_imports.go     math_random.yaml  use_of_weak_crypto_des.go   use_of_weak_rsa_key.go
bad_imports.yaml   ssl.go            use_of_weak_crypto_md5.go   use_of_weak_rsa_key.yaml
insecure_ssh.go    ssl.yaml          use_of_weak_crypto_rc4.go
insecure_ssh.yaml  tls.go            use_of_weak_crypto_sha1.go
math_random.go     tls.yaml          use_of_weak_crypto.yaml
```

The `use_of_weak_crypto` tests don't match the filename convention. This means `semgrep --test` isn't running these tests. Let's combine these test files into a single file that matches the naming convention :+1:
